src/arrayMethodJoin.js

```diff
@@ -4,8 +4,33 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    // if (typeof separator !== 'string') {
+    //   return this;
+    // }
+    if (this.length === 0) {
+      return '';
+    }
+
+    let result = `${this[0]}`;
+
+    if (this[0] === null) {
```

Comment:
This condition is not needed because you have already covered the case when 'this[0]' is null in the 'result' initialization.